### PR TITLE
Add ResponseListener.listenForResponseFromTransaction()

### PR DESCRIPTION
**Pull request by bolekk**

### CODE REVIEW

**Comment by bolekk** [comment]
@chudilka1 I fixed most issues except two:
1. Sonar tells me that "checkTimeout" should be const but it can't be const. 
2. Prettier wants to add one semicolon and eslint wants to remove it so I can't satisfy lint.

What should I do?

File: src/ResponseListener.ts [hunk 1]
```diff
@@ -55,7 +57,45 @@ export class ResponseListener {
...
+
+  public async listenForResponseFromTransaction(
+    txHash: string,
+    timeout = 3000000,
+    confirmations = 2,
+    checkInterval = 2000,
+  ): Promise<FunctionsResponse> {
+    return new Promise<FunctionsResponse>((resolve, reject) => {
+      (async () => {
+        let requestID: string = ''
```
Comment (KuphJr) [inline src/ResponseListener.ts:71]: nit: you shouldn't need to specify a type here. Typescript can infer the type based on the initial empty string value.

File: src/ResponseListener.ts [hunk 2]
```diff
@@ -55,7 +57,45 @@ export class ResponseListener {
...
+        // eslint-disable-next-line
```
Comment (KuphJr) [inline src/ResponseListener.ts:72]: What is being disabled here? Usually eslint-disble statementments disable a single warning:
example: `// eslint-disable-next-line no-implicit-any` (usually Copilot helps me do this automatically)

**Comment by justinkaseman** [comment]
>
>     2. Prettier wants to add one semicolon and eslint wants to remove it so I can't satisfy lint.
> 
> What should I do?

It looks like this repo is missing a prettier <> eslint integration, which is how you make them play nicely together.
Docs here: https://prettier.io/docs/en/integrating-with-linters.html
Let me know if you want me to iron this out.

**Comment by KuphJr** [comment]
> prettier.io/docs/en/integrating-with-linters.html

Thanks @justinkaseman! I'll take care of this now since I am the one who set up esLint to begin with.

**Comment by chudilka1** [comment]
> @chudilka1 I fixed most issues except two:
> 
> 1. Sonar tells me that "checkTimeout" should be const but it can't be const.
> 2. Prettier wants to add one semicolon and eslint wants to remove it so I can't satisfy lint.

@bolekk 
1. Is addressed from within Sonar (as won't fix with your comment)
2. Let's wait for @KuphJr to address linter <> prettier setup

File: src/ResponseListener.ts [hunk 3]
```diff
@@ -55,7 +57,45 @@ export class ResponseListener {
...
+        let checkTimeout: NodeJS.Timeout
+        const expirationTimeout = setTimeout(() => {
+          reject('Response not received within timeout period')
+        }, timeout)
+
+        const check = async () => {
```
[the remark this answers is not in the document]
Comment (bolekk) [reply src/ResponseListener.ts:78]: Yes. The old method has a similar problem - when we start listening after the response is emitted, we will miss it. It seems to be inherent to a functionsRouter.on() call. I'll add a note to README.

File: src/ResponseListener.ts (hunk 2 above)
Comment (bolekk) [reply src/ResponseListener.ts:72]: Lint claims this can be const but it can't, right? (I'm using it before setting)

Comment (bolekk) [reply src/ResponseListener.ts:72]: Adding "@typescript-eslint/prefer-as-const" doesn't work

**Pull request merged**